projector-server-core/src/main/kotlin/org/jetbrains/projector/server/core/ij/IdeaState.kt

```diff
@@ -61,12 +70,17 @@ public fun invokeWhenIdeaIsInitialized(
         if (ideaMainClassWithIdeaClassLoader != null) {  // null means we run with IDEA but it's not initialized yet
           val ideaClassLoader = ideaMainClassWithIdeaClassLoader.classLoader
 
-          if (isIdeaInProperState(ideaClassLoader)) {
-            onInitialized(ideaClassLoader)
+          if (onClassLoaderFetched != null) {
+            onClassLoaderFetched(ideaClassLoader)
+
+            logJobDone()
+            break
+          }
+
+          if (ProjectorClassLoaderSetup.ideaClassLoaderInitialized && isIdeaInProperState(ideaClassLoader)) {
+            onInitialized?.invoke(ideaClassLoader)
```

Comment:
Am I right that here we imply that `onClassLoaderFetched == null` and `onInitialized != null`? Then I suggest at least marking this in code:

```suggestion
            onInitialized!!(ideaClassLoader)
```

Minor: since now there is one more parameter in the function, maybe it's worth to finally describe its expected logic in the javadoc.

Reply:
Ah, yeah, it will loop infinitely if we use it with IDEA and both `onClassLoaderFetched` and `onInitialized` are null, good notice

Do you think doc for the function should be added in this PR?

Reply:
Yeah, let's add it now.

Reply:
Fixed and added javadoc